### PR TITLE
Add basic linting of META.yml files

### PR DESCRIPTION
**Pull request by jcscottiii**
Check that the file meets the correct format. The two keys in the META.yml are optional but no other keys should be present.

Additionally, this PR:
- Changes the lint command to use virtualenv

This is part of a multi PR plan
- PR 1 - Parse META.yml files - #43394
- PR 2 - Add lint of META.yml files (This PR) - #43395 
- PR 3 - Parse WEB_FEATURES.yml files - #43403
- PR 4 - Add lint of WEB_FEATURES.yml files - #43401

### CODE REVIEW

File: tools/lint/tests/test_file_lints.py [hunk 1]
```diff
@@ -776,6 +776,38 @@ def test_css_missing_file_in_css():
...
+    errors = check_file_contents("", "css/META.yml", io.BytesIO(code))
+    check_errors(errors)
+
+    assert errors == []
+
+def test_invalid_meta_file():
+    code = b"""\
+- test
+"""
+    # Check when the value is named correctly. It should find the error.
```
Comment (foolip) [inline tools/lint/tests/test_file_lints.py:794]:
```suggestion
    # Check when the file is named correctly. It should find the error.
```

File: tools/lint/tests/test_file_lints.py [hunk 2]
```diff
@@ -776,6 +776,38 @@ def test_css_missing_file_in_css():
...
+    check_errors(errors)
+
+    assert errors == [
+        ('INVALID-META-FILE',
+         'The META.yml file contains an invalid structure',
+         "css/META.yml",
+         None),
+    ]
+
+    # Check when the value is named incorrectly. It should not find the error.
```
Comment (foolip) [inline tools/lint/tests/test_file_lints.py:805]:
```suggestion
    # Check when the file is named incorrectly. It should not find the error.
```

File: tools/lint/rules.py [hunk 3]
```diff
@@ -339,6 +339,11 @@ class TentativeDirectoryName(Rule):
     to_fix = "rename directory to be called 'tentative'"
 
 
+class InvalidMetaFile(Rule):
+    name = "INVALID-META-FILE"
+    description = "The META.yml file contains an invalid structure"
```
Comment (foolip) [inline tools/lint/rules.py:344]: This lint error will happen both when the META.yml file cannot be parsed as YAML, and when it parses but doesn't pass the validation rules, right?

This description sounds most like it's about validation, not invalid YAML. Separating the two cases might make it easier to write the description, but for a single message how about...:

```suggestion
    description = "The META.yml is not a YAML file with the expected structure"
```

Comment (jcscottiii) [reply tools/lint/rules.py:344]: I see what you mean. Thanks!

**Review by foolip — APPROVED** [review]

**Pull request merged**